osu.Game/Overlays/Settings/Sections/Maintenance/GeneralSettings.cs

```diff
@@ -38,8 +39,12 @@ private void load(BeatmapManager beatmaps)
                     Text = "Delete ALL beatmaps",
                     Action = () =>
                     {
-                        deleteButton.Enabled.Value = false;
-                        Task.Run(() => beatmaps.DeleteAll()).ContinueWith(t => Schedule(() => deleteButton.Enabled.Value = true));
+                        Action deletion = delegate

```

Comment:
@Aergwyn Did you check this manually? It looks like the dialog is displaying after performing delete operation. Which is pointless.

Reply:
It's declared there and only executed when you accept. It's not run otherwise.